app/cdap/components/ConnectionList/index.tsx

```diff
@@ -0,0 +1,268 @@
+
+    setDataForTabs((prev): any => {
+      const tempData = [...prev];
+      tempData.push({
+        data: [],
+        showTabs: false,
+        selectedTab: '',
+        isSearching: false,
+      });
+      tempData[index + 1][`data`] = connections;
+      tempData[index + 1][`showTabs`] = true;
+      tempData[index + 1][`selectedTab`] = null;
+      return tempData.slice(0, index + 2);
+    });
+    toggleLoader(false);
+  };
+
+  const fetchEntities = async (connectionId, url = pathFromUrl) => {
+    const pathDesired = url ? url : pathFromUrl;
+    const entitiesPromise = exploreConnection({
+      connectionid: connectionId,
+      path: pathDesired,
+    });
+    return entitiesPromise.then((values) => {
+      return values;
+    });
+  };
+
+  const selectedTabValueHandler = (entity: any, index: number) => {
+    toggleLoader(true);
+    setDataForTabs((currentData): any => {
+      let newData = [...currentData];
+      newData[index].selectedTab = entity.name;
+      newData = newData.map((each) => {
+        return {
+          data: each.data,
+          showTabs: each.showTabs,
+          selectedTab: each.selectedTab,
+          isSearching: false,
+        };
+      });
+      if (index === 0) {
+        getCategorizedConnectionsforSelectedTab(entity.name, index);
+      } else if (index === 1) {
+        fetchEntities(entity.name).then((res) => {
+          setDataForTabs((prev): any => {
+            const tempData = [...prev];
+            tempData.push({
+              data: [],
+              showTabs: false,
+              selectedTab: '',
+              isSearching: false,
+            });
+            tempData[index + 1][`data`] = res.entities;
+            tempData[index + 1][`showTabs`] = true;
+            tempData[index + 1][`selectedTab`] = null;
+            tempData[index + 1][`isSearching`] = false;
+            return tempData.slice(0, index + 2);
+          });
+          toggleLoader(false);
+        });
+      } else {
+        if (entity.canBrowse) {
+          fetchEntities(dataForTabs[1].selectedTab, entity.path).then((res) => {
+            setDataForTabs((prev): any => {
+              const tempData = [...prev];
+              tempData.push({
+                data: [],
+                showTabs: false,
+                selectedTab: '',
+                isSearching: false,
+              });
+              tempData[index + 1][`data`] = res.entities;
+              tempData[index + 1][`showTabs`] = true;
+              tempData[index + 1][`selectedTab`] = null;
+              tempData[index + 1][`isSearching`] = false;
+              return tempData.slice(0, index + 2);
+            });
+            toggleLoader(false);
+          });
+        }
+      }
+      return newData;
+    });
+  };
+
+  React.useEffect(() => {
+    getConnectionsTabData();
+  }, []);
+
+  React.useEffect(() => {
+    setDataForTabs((prev) => {
+      const temp = [...prev];
```

Comment:
why the spread here? do you need to not modify the previous version? could you put a comment saying that if thats true?

Reply:
Done